src/sentry/auth/access.py

```diff
@@ -182,6 +182,7 @@ def __init__(
         sso_is_valid,
         requires_sso,
         permissions=None,
+        role=None,
```

Comment:
Would be good to add tests around this property to capture how it is supposed to work for each access factory path.